src/core/thread/meshcop_dataset_manager.cpp

```diff
@@ -268,6 +270,37 @@ void DatasetManager::HandleSet(Coap::Header &aHeader, Message &aMessage, const I
     return;
 }
 
+void DatasetManager::HandleGet(void *aContext, Coap::Header &aHeader, Message &aMessage,
+                               const Ip6::MessageInfo &aMessageInfo)
+{
+    DatasetManager *obj = static_cast<DatasetManager *>(aContext);
+    obj->HandleGet(aHeader, aMessage, aMessageInfo);
+}
+
+void DatasetManager::HandleGet(Coap::Header &aHeader, Message &aMessage, const Ip6::MessageInfo &aMessageInfo)
+{
+    Tlv tlv;
+    uint16_t offset = aMessage.GetOffset();
+    uint8_t tlvs[Dataset::kMaxSize];
+    uint8_t length = 0;
+
+    while (offset < aMessage.GetLength())
+    {
+        aMessage.Read(offset, sizeof(tlv), &tlv);
+        length = tlv.GetLength();
+
+        if (tlv.GetType() == Tlv::kGet)
+        {
+            aMessage.Read(offset + sizeof(Tlv), length, tlvs);
+            break;
+        }
+
+        offset += sizeof(tlv) + tlv.GetLength();
+    }
+
+    SendGetResponse(aHeader, aMessageInfo, tlvs, length);
```

Comment:
Yes, you are correct.  I meant to say, the `length` should be set to 0.  However, `length` may not be set to zero if a `Get TLV` is not found.